### PR TITLE
feat(animations): back button and view animationsa

### PR DESCRIPTION
**Pull request by mhartington**
#### Short description of what this resolves:
Fixes animations for back buttons and pages

### CODE REVIEW

File: packages/core/src/components/nav/nav-utils.ts [hunk 1]
```diff
@@ -70,17 +70,15 @@ export function updateZIndex(viewController: ViewController, newZIndex: number)
...
+export function toggleHidden(element: HTMLElement, shouldBeHidden: boolean) {
+    element.hidden = shouldBeHidden;
 }
 
-export function canNavGoBack(nav: Nav) {
+export function canNavGoBack(nav: Nav, view?: ViewController) {
   if (!nav) {
     return false;
   }
-  return !!nav.getPrevious();
```
Comment (manucorporat) [inline packages/core/src/components/nav/nav-utils.ts]: this function returns a boolean, we need the cast to boolean, right?

File: packages/core/src/components/nav/nav.scss [hunk 2]
```diff
@@ -13,7 +13,10 @@ ion-nav {
   width: 100%;
   height: 100%;
 
+  background-color: black;
```
Comment (manucorporat) [inline packages/core/src/components/nav/nav.scss]: ?

File: packages/core/src/components/nav/transitions/transition.ios.ts [hunk 3]
```diff
@@ -11,174 +11,198 @@ const CENTER = '0%';
...
 const SHOW_BACK_BTN_CSS = 'show-back-button';
 
-// TODO - make sure this uses the `ion-page` logic from the md transition
-// DO this later since the transition is broke as a joke anyway
-// Dan B 1/9/2018
-
 export function buildIOSTransition(rootTransition: Transition, enteringView: ViewController, leavingView: ViewController, opts: AnimationOptions): Promise<Transition> {
+  const componentReadyPromise: Promise<any>[] = [];
+  // Let makes sure everything is hydrated and ready to animate
+  if (enteringView) {
```
Comment (manucorporat) [inline packages/core/src/components/nav/transitions/transition.ios.ts]: I think this promise should not be here, but outside

File: packages/core/src/components/nav/nav-utils.ts (hunk 1 above)
Comment (danbucholtz) [reply packages/core/src/components/nav/nav-utils.ts]: @mhartington - I think Manu's right

**Pull request merged**